dae_testing/homework_example/HW8_3a.py

```diff
@@ -27,6 +27,39 @@ def make_model():
 
     m.obj = Objective(expr=m.x3[m.tf])
 
+    def _init(m):
+        yield m.x1[0] == 0
+        yield m.x2[0] == -1
+        yield m.x3[0] == 0
+    m.init_conditions = ConstraintList(rule=_init)
+    return m
+
+def Constraint_definitions(m,t):
+    def _x1dot(m, t):
+        if t in m.non_coll_disc_pts: 
+            return Constraint.Skip
+        else:
+            return m.dx1dt[t] == m.x2[t]
+    m.x1dot = Constraint(m.t, rule=_x1dot)
+
+    def _x2dot(m, t):
+        if t in m.non_coll_disc_pts: 
+            return Constraint.Skip
+        else:
+            return m.dx2dt[t] == -m.x2[t] + m.u[t]
+    m.x2dot = Constraint(m.t, rule=_x2dot)
+
+    def _x3dot(m, t):
+        if t in m.non_coll_disc_pts: 
+            return Constraint.Skip
+        else:
+            return m.dx3dt[t] == m.x1[t]**2 + m.x2[t]**2 + 0.005*m.u[t]**2
+    m.x3dot = Constraint(m.t, rule=_x3dot)
+
+#Constrain_definitions ignores these constraints at non-collocation, 
+#discretization points
+

```

Comment:
Here we're assuming that the model has an attribute called `non_coll_disc_pts`, at which we should skip these constraints. I think a more general and user-friendly way to do this is to construct the model in the same way as before, then identify constraints at these points and deactivate them. This way we can "remove" constraints at non-collocation finite element points in a user's preexisting model without intrusive modification. Of course, we don't want to deactivate discretization or continuity equations, so we'll have to use constraint names to check for these (e.g. check `name.ends_with("_disc_eq")`.